cpp/include/DataStorm/Node.h

```diff
@@ -43,8 +43,31 @@ namespace DataStorm
          *
          * @param communicator The Ice communicator used by the topic factory for its configuration and
          *                     communications.
+         * @param callbackExecutor An optional callback executor used to execute user callbacks, if no callback executor
+         *                         is provided the Node will use the default callback executor that executes callback in
+         *                         a dedicated thread.
          */
-        Node(Ice::CommunicatorPtr communicator);
+        Node(
+            Ice::CommunicatorPtr communicator,
+            std::function<void(std::function<void()> call)> callbackExecutor = nullptr);
+
+        /**
```

Comment:
I would reorder the constructor from most commonly used to least commonly used.

Reply:
I guess the most common would be

```
Node(arg, argv);
Node(communicator);
Node(executor, arg, argv);
```